crates/solver/src/settlement_submission/submitter/public_mempool_api.rs

```diff
@@ -56,24 +56,31 @@ impl TransactionSubmitting for PublicMempoolApi {
                             node.web3.eth().send_raw_transaction(bytes.0.into()).await
                         }
                     };
+                    let can_broadcast = node.can_broadcast();
 
-                    (label, result)
+                    (label, result, can_broadcast)
                 }
                 .boxed()
             })
             .collect::<Vec<_>>();
 
         let mut errors = vec![];
         loop {
-            let ((label, result), _, rest) = futures::future::select_all(futures).await;
+            let ((label, result, can_broadcast), _, rest) =
+                futures::future::select_all(futures).await;
             match result {
                 Ok(tx_hash) => {
                     super::track_submission_success(&label, true);
-                    tracing::debug!(%label, "created transaction with hash: {:?}", tx_hash);
-                    return Ok(TransactionHandle {
-                        tx_hash,
-                        handle: tx_hash,
-                    });
+                    if can_broadcast {
+                        tracing::debug!(%label, "created transaction with hash: {:?}", tx_hash);
+                        return Ok(TransactionHandle {
+                            tx_hash,
+                            handle: tx_hash,
+                        });
+                    } else {
+                        tracing::debug!(%label, "non-broadcasting node reports transaction creation with hash: {:?}", tx_hash);
+                        futures = rest;
+                    }
```

Comment:
It might be even simpler to fire and forget by `tokio::task:spawn`ing the requests to notification nodes. That way we don't need to handle the response at all. But also that might induce more control flow changes so not sure.

Reply:
I implemented this [here](https://github.com/cowprotocol/services/pull/1178).